improver/blending/spatial_weights.py

```diff
@@ -289,39 +152,156 @@ def create_template_slice(cube_to_collapse, blend_coord):
         # Return slice template
         return first_slice
 
-    def process(self, cube_to_collapse, one_dimensional_weights_cube, blend_coord):
+    def _normalise_initial_weights(self, weights):
+        """Normalise weights so that they add up to 1 along the blend dimension
+        at each spatial point.  This is different from the normalisation that
+        happens after the application of fuzzy smoothing near mask boundaries.
+        Modifies weights cube in place.
+
+        Args:
+            weights (iris.cube.Cube):
+                3D weights containing zeros for masked points, but before
+                fuzzy smoothing
+        """
+        weights_sum = np.sum(weights.data, axis=self.blend_axis)
+        weights_sum = np.broadcast_to(weights_sum, weights.shape)
```

Comment:
I don't think this line is required, it essentially uses vectorisation to do the divide (which can be done without the memory view).